apis/python/src/tiledbsoma/_dataframe.py

```diff
@@ -427,7 +429,30 @@ def resize_soma_joinid_shape(self, newshape: int) -> None:
         domain: in that case please call ``tiledbsoma_upgrade_domain`` (WIP for
         1.15).
         """
-        self._handle._handle.resize_soma_joinid_shape(newshape)
+        if check_only:
+            return cast(
+                Tuple[bool, str],
+                self._handle._handle.can_resize_soma_joinid_shape(newshape),
+            )
+        else:
+            self._handle._handle.resize_soma_joinid_shape(newshape)
+            return (True, "")
+
+    def upgrade_soma_joinid_shape(
+        self, newshape: int, check_only: bool = False
+    ) -> Tuple[bool, str]:
+        """This is like ``upgrade_domain``, but it only applies the specified domain
+        update to the ``soma_joinid`` index column. Any other index columns have their
+        domain set to match the maxdomain. If the ``soma_joinid`` column is not an index
+        column at all, then no actiong is taken."""
```

Comment:
For all functions that return a Tuple[bool, str], please include in the docstrings what the `bool` and `str` represent (e.g., the success status of the operation and a detailed message). 

We could also add a type alias `ValidationResult = Tuple[bool, str]` (or a better name) in `_types.py` which would help further clarify what the return values represent.

Reply:
@nguyenv second commit https://github.com/single-cell-data/TileDB-SOMA/pull/3140/commits/8c7752a2e21fed9b70b9755010fcd4705d4904fc